src/libmongoc/tests/unified/operation.c

```diff
@@ -2761,6 +2761,46 @@ operation_assert_number_connections_checked_out (test_t *test,
    return true;
 }
 
+static bool
+operation_rename (test_t *test,
+                  operation_t *op,
+                  result_t *result,
+                  bson_error_t *error)
+{
+   // First validate the arguments
+   const char *object = op->object;
+   bson_parser_t *bp = bson_parser_new ();
+   char *new_name;
+   bson_parser_utf8 (bp, "to", &new_name);
+   bool parse_ok = bson_parser_parse (bp, op->arguments, error);
```

Comment:
The parser copies the fields it parses. `new_name` will need to be freed with `bson_free (new_name);`. Or, the parser can be freed with `bson_parser_destroy_with_parsed_fields`.